test/sqllogictest/try_parse_monotonic_iso8601_timestamp.slt

```diff
@@ -0,0 +1,249 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+# Real-world example
+query T
+select try_parse_monotonic_iso8601_timestamp('2015-09-18T23:56:04.123Z');
```

Comment:
The timezone is currently fixed to `Z`, right? Therefore, I assume it is to expect that `+2` fails.